File: Java_Codes/15. Memory Management/StackHeapExample.java
//Stack memory manages method execution while heap memory stores large 
//data sets such as user information in a web application.
public class StackHeapExample {

    public static void demonstrateStackHeap() {
        // Stack allocation
        int localVariable = 42;
        System.out.println("Local variable (Stack): " + localVariable);

        // Heap allocation
        MyObject myObject = new MyObject("Heap Object");
        System.out.println("Object stored in Heap: " + myObject.getName());
    }

    static class MyObject {
        private String name;

        public MyObject(String name) {
            this.name = name;
        }

        public String getName() {
            return name;
        }
    }

    public static void main(String[] args) {
        demonstrateStackHeap();
    }
}

File: Python_Codes/15. Memory Management/StackHeapExample.py
class MyObject:
    def __init__(self, name):
        self.name = name  # Stored in heap

    def get_name(self):
        return self.name

def demonstrate_stack_heap():
    # Stack allocation (local variable)
    local_variable = 42
    print("Local variable (Stack):", local_variable)

    # Heap allocation (object)
    my_object = MyObject("Heap Object")
    print("Object stored in Heap:", my_object.get_name())

if __name__ == "__main__":
    demonstrate_stack_heap()
